pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -442,14 +442,22 @@ private void takeSnapshotByTimeout() {
                 takeSnapshotIntervalTime, TimeUnit.MILLISECONDS);
     }
 
+    private PositionImpl getPreviousPosition(PositionImpl position) {
```

Comment:
remove the method and use ` topic.getManagedLedger()).getPreviousPosition(position);` directly is enough.

Reply:
fixed.